gcc/config/aarch64/aarch64.h

```diff
@@ -503,7 +512,11 @@ enum class aarch64_feature : unsigned char {
    uses alloca.  */
 #define EXIT_IGNORE_STACK	(cfun->calls_alloca)
 
+#ifdef TARGET_64BIT_MS_ABI
 #define STATIC_CHAIN_REGNUM		R17_REGNUM
+#else
+#define STATIC_CHAIN_REGNUM		R18_REGNUM
```

Comment:
I do not understand why we change STATIC_CHAIN_REGNUM for not MS_ABI

Reply:
it looks for me as a fix on CALL_USED_X18 regression which was introduced earlier in this PR

Reply:
STATIC_CHAIN_REGNUM defines the register used for passing a function's static chain pointer. X18 cannot be used for this purpose on Windows because it is reserved for the TEB.

Reply:
I know regarding x18 on WINDOWS ABI. I checked the history of changes. It was originally x18 for aarch64